lib/resourceapply/cv.go

```diff
@@ -15,36 +17,46 @@ import (
 	"k8s.io/utils/pointer"
 )
 
-func ApplyOperatorStatus(client osclientv1.ClusterOperatorsGetter, required *osv1.ClusterOperator) (*osv1.ClusterOperator, bool, error) {
+func ApplyClusterOperatorStatus(client osclientv1.ClusterOperatorsGetter, required *osv1.ClusterOperator) (*osv1.ClusterOperator, bool, error) {
 	if required.Status.Extension.Raw != nil && required.Status.Extension.Object != nil {
 		return nil, false, fmt.Errorf("both extension.Raw and extension.Object should not be set")
 	}
 	existing, err := client.ClusterOperators(required.Namespace).Get(required.Name, metav1.GetOptions{})
 	if errors.IsNotFound(err) {
 		actual, err := client.ClusterOperators(required.Namespace).Create(required)
+		if err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
if we couldn't find the object why do this check... i'm probably missing something here?

Reply:
for status you can't update the status at the same time as creation (creation and update only allow spec changes, you have to call update status).  So this is creating whatever the object is, then applying the status.  I think we could change these methods around to bail out, but since ClusterOperators is primarily a status object this seemed the closest to the intent of apply:  "make sure this object exists with this status".  We always pass in a valid ClusterOperator here.

Also, if someone deletes the ClusterOperator out from under us we just recreate it (we're the object owner).

Reply:
hmm, i didn't realize that `Create&Update` don't change `.status`. :thinking: